.github/workflows/tox.yml

```diff
@@ -35,6 +35,7 @@ jobs:
         - TOXENV: build-dists,metadata-validation
     env:
       TOX_PARALLEL_NO_SPINNER: 1
+      PYTEST_ADDOPTS: -n auto
```

Comment:
Move it to the pytest config. Everyone needs it

Reply:
That is not true, if I put it inside pytest.ini it will be impossible for me to run tests without parallel mode, within or outside tox. Only settings that are really universal should go there, and ability to decide between serial and parallel is not one of them.

If you want, there is another option I used in the past, to add it to tox.ini like:
```
setenv =
    PYTEST_ADDOPTS={env:PYTEST_ADDOPTS:-n auto}
```

This means tox will run with auto unless user defines his own set of pytest preferences, which would override it, allowing a developer to bypass it **without altering tracked files**.

Still, AFAIK, `pytest.ini` settings do not allow much flexibility

Reply:
> That is not true, if I put it inside pytest.ini it will be impossible for me to run tests without parallel mode, within or outside tox.

This is incorrect. I do it all the time when I need to. `tox` has `{posargs}` set correctly and so it's super easy to run one test with parallelism using the following command:

```console
$ tox -e ansible29-py38 -- test/TestMatchError.py::test_matcherror_compare_invalid -n 0
```

I just copied it from my zsh history, this is what I use very often. The trick is to remember that whatever you pass after `--` gets placed in place of `{posargs}` and since it's in the end of pytest invocation command, anything you pass there will override any previous options, regardless of whether they are passed via CLI, `PYTEST_ADDOPTS` env var or `addopts` config option.

Reply:
Pytest itself has it set up like this: https://github.com/pytest-dev/pytest/blob/master/tox.ini#L22. Admittedly, their config is more complicated than we need but this is an idiom that I see and rely on probably in every project that I ever saw on GitHub or elsewhere using tox+pytest.

Reply:
More examples:

- https://github.com/pypa/setuptools/blob/master/tox.ini#L24
- https://github.com/pypa/twine/blob/master/tox.ini#L18
- https://github.com/pypa/virtualenv/blob/master/tox.ini#L37

Reply:
Maybe I was not clear: as a developer, I do not want to be forced to add `-n 0` when I run pytest on ansible-lint in order to get serial execution of tests.

I am aware that `pytest {posargs:-n auto}` inside tox.ini is another alternative, one that I used in other projects too.

As we are again blocked around personal preferences, I would like to get another opinion in order to make a decision and move on.

Reply:
FWIW what you added does not work.

Reply:
Besides, the point of this is to make developers case about tests not influencing each other's state and clean things up correctly.